### PR TITLE
feat: add ManabiS3Provider

### PR DESCRIPTION
**Pull request by open-dynaMIX**

### CODE REVIEW

File: manabi/filesystem.py [hunk 1]
```diff
@@ -142,24 +150,167 @@ def __init__(
...
+                cb_hook_config=self._cb_hook_config,
+            )
+        except ClientError as ex:
+            if ex.response["Error"]["Code"] == "NoSuchKey":
+                # File does not exist
+                return None
+
+    def file_exists(self, path: Path):
+        # Is handled implicitly in `get_file_resource()`
+        return True
```
Comment (ganwell) [inline manabi/filesystem.py:316]: My need to conform the protocol (even though not strictly needed), would probably make do something like:

```python
return bool(self.s3.get_object(Bucket=self.bucket_name, Key=path))
```

any downsides to this? (I haven't checked the interface get_object, maybe it throws exceptions if the file does not exists). Is it a performance thing? In that case we might cache s3-object?

**Review by ganwell — APPROVED** [review]
The implementation look very clean, thanks a lot. I know this is currently just a draft, but since I am away next week I looked at it already.

👍

File: manabi/filesystem.py (hunk 1 above)
Comment (open-dynaMIX) [reply manabi/filesystem.py]: I've refactored the `file_exists()` logic into the `get_file_resource()` method.

Comment (ganwell) [reply manabi/filesystem.py]: I read ` def do_PROPFIND` in wsgi-dav and I see now that this works properly. But I did not find `file_exists()` in wsgi-dav at all. I think it got removed at some point. So you should be able to remove `def file_exists()` completely.

**Comment by ganwell** [comment]
I think pytest will proof whether you can remove `def file_exists()` or not, so consider this approved. Unfortunately I cannot approve without resolving the `def file_exists()` thread. I'll have a look into this PR tomorrow evening.

Great work thanks.

File: manabi/filesystem.py (hunk 1 above)
Comment (open-dynaMIX) [reply manabi/filesystem.py]: `file_exists()` was an approach by me, to handle different types of paths (fs vs s3). But it's already removed, as per my commend above:

> I've refactored the file_exists() logic into the get_file_resource() method.

**Pull request merged**